core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -606,64 +606,58 @@ object Parser {
         }
       }
 
+    private def mergeInRange(irs: List[InRange]): List[InRange] = {
+      @annotation.tailrec
+      def merge(rs: List[InRange], aux: Chain[InRange] = Chain.empty): Chain[InRange] =
+        rs match {
+          case x :: y :: rest =>
+            if (y.lower.toInt > x.upper.toInt + 1) merge(y :: rest, aux :+ x)
```

Comment:
It's not clear to me this is merging the regions correctly. You aren't consulting y.upper.

Could we copy the old code out to a test and verify that the behavior is the same?

Reply:
Sure thing. Added a test with it.